### PR TITLE
Multi-select component

### PR DESCRIPTION
**Pull request by dsgraham**
Adds a new component that behaves like a select list, but is composed of a checkbox group.

### CODE REVIEW

File: lib/voom/presenters/dsl/components/multi_select.rb [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+            end
+
+            def checkbox(**attributes, &block)
+              return @checkbox if locked?
+              @checkbox = Components::Checkbox.new(parent: self,
+                                                   **attributes,
+                                                   &block)
+            end
+
+            def _selected?
```
Comment (rx) [inline lib/voom/presenters/dsl/components/multi_select.rb:50]: Any reason this has the `_` on the method name?

File: views/mdc/assets/js/components/multi-select.js [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+export class VMultiSelect {
+
+  constructor(root, element) {
+    this.vComponent = root.vComponent;
+    this.element = element;
+    this.hidden_on_create = null;
+    this.mutationObserver = null;
+    this.setEventListeners();
+    this.setLabelHandlers();
+    // this.setMutationHandler();
```
Comment (rx) [inline views/mdc/assets/js/components/multi-select.js:20]:
```suggestion
```

File: views/mdc/assets/js/components/multi-select.js [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+    document.addEventListener('click', createCloseHandler(this.element));
+  }
+
+  setLabelHandlers() {
+    createValueDescriptionHandler(this.element);
+    setCurrentValueDescription(this.element);
+    setLabelNotch(this.element);
+  }
+
+  // Attempt to re-render when a hidden component becomes visible
```
Comment (rx) [inline views/mdc/assets/js/components/multi-select.js:34]: Same as above ... remove commented out code.

File: views/mdc/components/multi_select.erb [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+      <div class="mdc-notched-outline__leading"></div>
+      <div class="mdc-notched-outline__notch">
+        <label class="mdc-floating-label <%= 'mdc-floating-label--float-above' if float_label %>"><%= comp.label %></label>
+      </div>
+      <div class="mdc-notched-outline__trailing"></div>
+    </div>
+  </div>
+  <%= erb :"components/shared/hint_error_display", :locals => {comp: comp} %>
+  <%= erb :"components/tooltip", :locals => {comp: comp.tooltip, parent_id: comp.id} %>
+<% end %>
```
Comment (rx) [inline views/mdc/components/multi_select.erb:31]:
```suggestion
<% end %>

```

**Review by rx — CHANGES_REQUESTED** [review]
Demo error: http://localhost:9292/multi_selects
```
NoMethodError at /multi_selects
undefined method `checkbox' for #<Voom::Presenters::DSL::Components::Select::Option:0x00007f93461b5388>
file: multi_select.erb location: block in __tilt_70135244506840 line: 18

BACKTRACE
(expand)

JUMP TO: GET POST COOKIES ENV
/Users/russelledens/dev/rx/presenters/views/mdc/components/multi_select.erb in block in __tilt_70135244506840
        <%= erb :"components/checkbox", :locals => {comp: option.checkbox, class_name: 'v-multi-select--option'} %>
/Users/russelledens/dev/rx/presenters/views/mdc/components/multi_select.erb in each
      <% comp.options.each do |option| %>
/Users/russelledens/dev/rx/presenters/views/mdc/components/multi_select.erb in __tilt_70135244506840
      <% comp.options.each do |option| %>
```

File: lib/voom/presenters/dsl/components/multi_select.rb (hunk 1 above)
Comment (dsgraham) [reply lib/voom/presenters/dsl/components/multi_select.rb:50]: That just got copied over from the Select::Option class. Neither of them appear to be used anywhere, so I'll look into removing them.

**Review by rx — APPROVED** [review]

**Review by rx — CHANGES_REQUESTED** [review]
Seeing a CSS issue in the demo:
Image of CS Issue with Multi Selects

**Comment by dsgraham** [comment]
> Seeing a CSS issue in the demo:

Yeah, I'm still working on that one.

**Review by rx — APPROVED** [review]

**Pull request merged**